extensions-api/src/main/scala/pl/touk/nussknacker/engine/testmode/TestProcess.scala

```diff
@@ -5,68 +5,55 @@ import pl.touk.nussknacker.engine.api.{Context, ContextId}
 
-    def updateNodeResult(nodeId: String, context: Context) = {
-      copy(nodeResults =
-        nodeResults + (nodeId -> (nodeResults.getOrElse(nodeId, List()) :+ NodeResult(toResult(context))))
-      )
+    def updateNodeResult(nodeId: String, context: Context): TestResults = {
+      copy(nodeResults = nodeResults + (nodeId -> (nodeResults.getOrElse(nodeId, List()) :+ NodeResult(context))))
     }
 
-    def updateExpressionResult(nodeId: String, context: Context, name: String, result: Any) = {
-      val invocationResult = ExpressionInvocationResult(context.id, name, variableEncoder(result))
+    def updateExpressionResult(nodeId: String, context: Context, name: String, result: Any): TestResults = {
+      val invocationResult = ExpressionInvocationResult(context.id, name, result)
       copy(invocationResults =
         invocationResults + (nodeId -> addResults(invocationResult, invocationResults.getOrElse(nodeId, List())))
       )
     }
 
-    def updateExternalInvocationResult(nodeId: String, contextId: ContextId, name: String, result: Any) = {
-      val invocation = ExternalInvocationResult(contextId.value, name, variableEncoder(result))
+    def updateExternalInvocationResult(nodeId: String, contextId: ContextId, name: String, result: Any): TestResults = {
+      val invocation = ExternalInvocationResult(contextId.value, name, result)
       copy(externalInvocationResults =
         externalInvocationResults + (nodeId -> (externalInvocationResults.getOrElse(nodeId, List()) :+ invocation))
       )
     }
 
-    def updateExceptionResult(exceptionInfo: NuExceptionInfo[_ <: Throwable]): TestResults[T] =
+    def updateExceptionResult(exceptionInfo: NuExceptionInfo[_ <: Throwable]): TestResults =
       copy(exceptions = exceptions :+ exceptionInfo)
 
     // when evaluating e.g. keyBy expression can be invoked more than once...
     // TODO: is it the best way to handle it??
     private def addResults(
-        invocationResult: ExpressionInvocationResult[T],
-        resultsSoFar: List[ExpressionInvocationResult[T]]
+        invocationResult: ExpressionInvocationResult,
+        resultsSoFar: List[ExpressionInvocationResult]
     ) = resultsSoFar.filterNot(res =>
       res.contextId == invocationResult.contextId && res.name == invocationResult.name
     ) :+ invocationResult
 
-    private def toResult(context: Context): ResultContext[T] =
-      ResultContext(context.id, context.variables.map { case (k, v) => k -> variableEncoder(v) })
-
   }
 
-  /*
-    We have to be careful not to put too much into results, as they are serialized to JSON.
-   */
-  case class NodeResult[T](context: ResultContext[T]) {
-
-    def variableTyped[U <: T](name: String): Option[U] = context.variableTyped(name)
-
+  /**
+    * We have to be careful not to put too much into results, as they are serialized to JSON.
+    * TODO: Does it really have any sense? Why not use Context instead of it?
+    */
+  case class NodeResult(context: Context) {
```

Comment:
Maybe it's time to get rid of it? WDY @arkadius?

Reply:
I aggree